futures-executor/src/enter.rs

```diff
@@ -22,11 +22,13 @@ pub struct EnterError {
 /// executor.
 ///
 /// Executor implementations should call this function before blocking the
-/// thread.
+/// thread. Doing so ensures that executors aren't accidentally invoked in a
```

Comment:
"before blocking the thread [for the first time]."

Reply:
Also, it would be good to include a comment about keeping the `Enter` result around until you're *done* blocking the task.